File: benchmark/list.py
# List growth, indexed reads, and numeric accumulation.
import time

t0 = time.perf_counter()
entries = []
i = 0

while i < 100_000:
    entries.append(i * 3)
    i += 1

total = 0
i = 0
while i < len(entries):
    total += entries[i]
    i += 1

print((time.perf_counter() - t0) * 1000)

File: benchmark/list.lua
-- List growth, indexed reads, and numeric accumulation.
local entries = {}
local i = 0

while i < 100000 do
    entries[#entries + 1] = i * 3
    i = i + 1
end

local total = 0
i = 1
while i <= #entries do
    total = total + entries[i]
    i = i + 1
end
